feat(docs): add glossary page (#494)
input-output-hk/hydra: docs/docs/getting-started/glossary.md

# Glossary

### Contestation Period

> The time window between closing the head and `fan out` allowed to `contest`.

Reviewer: This is a bit unclear 🤔

Reviewer: What about: "The time window between closing of the head and the last possible contest from a head participant."

### Hydra Head
A running instance of the hydra head protocol. See a reference in [essential cardano](https://www.essentialcardano.io/glossary/hydra-head) for more details.

### Hydra Network
One or more hydra nodes inter-connected.

### Hydra Node

> Essential building block and executable piece of software that is used to form a [hydrahead network.

Reviewer: ```suggestion
Essential building block and executable piece of software that is used to form a Hydra network.
```

### Isomorphic
Reuses the existing smart contract system and code for offchain computations. This guarantees that onchain and offchain semantics coincide, leading to significant engineering simplifications.

### Layer 2
Layer 2 refers to a secondary framework or protocol that is built on top of an existing blockchain system. See this [page](../../core-concepts/layer-two) for more details.

### Marker Datum

> Its a [datum](https://docs.cardano.org/glossary/#datum) used for the sole purpose of identifying a [UTxO](https://docs.cardano.org/glossary/#utxo)

Reviewer: ```suggestion
Its a [datum](https://docs.cardano.org/glossary/#datum) used for the sole purpose of identifying a [UTxO](https://docs.cardano.org/glossary/#utxo).
```

### Participation Token

> An on-chain token entitling a party with the right to participate in the hydra head.

Reviewer: ```suggestion
An on-chain token that entitles a party to participate in the Hydra head associated with the token.
```

### Party

> A participant or logical member of the head network, identified by a pair of verification keys one for hydra and another one for cardano.

Reviewer: ```suggestion
A participant or logical member of a Hydra head, identified by a pair of verification keys one for the Hydra network and another one for the Cardano network.
```

### Peer

> A member of the head network, identified by an IP address and a TCP port.

Reviewer: ```suggestion
A member of a Hydra network, identified by an IP address and a TCP port.
```

> ### Protocol Fuel

Reviewer: ```suggestion
### Fuel UTxO
```

> A single [UTxO](https://docs.cardano.org/glossary/#utxo) used to pay for hydra head [transaction](https://docs.cardano.org/glossary/#transaction) [fees](https://docs.cardano.org/glossary/#fee)

Reviewer: ```suggestion
[UTxO](https://docs.cardano.org/glossary/#utxo) pieces used to pay for Hydra head [transaction](https://docs.cardano.org/glossary/#transaction) [fees](https://docs.cardano.org/glossary/#fee) happening on the base layer (i.e. the Cardano network).
```

### Reference script
See this [page](./quickstart#reference-scripts) for more details.

### Rollback

> Is actually a fork, and its a multi-step process between the head and L1. See this [page](../../core-concepts/rollbacks) for more details.

Reviewer: ```suggestion
The manifestation of a chain switch on the base layer, causing the recent history to change. It is a multi-step process between the head and L1. See this [page](../../core-concepts/rollbacks) for more details.
```

### Signing key
See a reference in [cardano glossary](https://docs.cardano.org/glossary/#keypair) for more details.

### Snapshot

> Some state of a hydra head multi signed by all parties.

Reviewer: ```suggestion
Some state of a Hydra head; When multi-signed by all parties, a snapshot is said "confirmed".
```

### State Channel

> See this [page](../../core-concepts/layer-two#state-channels) for more details.

Reviewer: ```suggestion
A common kind of layer two solution. See this [page](../../core-concepts/layer-two#state-channels) for more details.
```

### State Thread Token

> An on-chain token representing the state of a hydra head.

Reviewer: ```suggestion
An on-chain token representing the state of a Hydra head.
```
